x-pack/plugins/security_solution/cypress/e2e/detection_alerts/alerts_details.cy.ts

```diff
@@ -131,4 +133,60 @@ describe('Alert details flyout', () => {
       cy.get('[data-test-subj="formatted-field-kibana.alert.url"]').should('have.text', alertUrl);
     });
   });
+
+  describe('Localstorage management', { testIsolation: false }, () => {
+    before(() => {
+      cleanKibana();
+      esArchiverLoad('query_alert');
+      login();
+      visit(ALERTS_URL);
+      waitForAlertsToPopulate();
+    });
+
+    beforeEach(() => {
+      expandFirstAlert();
+    });
+
+    it('should not reopen the flyout when navigating away from the alerts page and returning to it', () => {
+      cy.get(OVERVIEW_RULE).should('be.visible');
+      goToRuleDetails();
+      cy.wait(500); // Wait for the localStorage update to take place
+      visit(ALERTS_URL);
+      cy.get(OVERVIEW_RULE).should('not.exist');
+    });
+
+    it('should store the flyout state in localstorage', () => {
+      cy.get(OVERVIEW_RULE).should('be.visible');
+      cy.wait(500); // Wait for the localStorage update to take place
+      cy.getAllLocalStorage().then((storage) => {
+        const securityDataTable = getLocalstorageEntryAsObject(storage, 'securityDataTable');
+        expect(securityDataTable?.['alerts-page']?.expandedDetail?.query?.panelView).to.eql(
+          'eventDetail'
+        );
+      });
+    });
+
+    it('should remove the flyout details from local storage when closed', () => {
+      cy.get(OVERVIEW_RULE).should('be.visible');
+      closeAlertFlyout();
+      cy.wait(500); // Wait for the localStorage update to take place
```

Comment:
No, it's because of how we handle the localStorage updates: https://github.com/elastic/kibana/blob/8c336b24d4e741325fb2c0522dd66808bb382bba/x-pack/plugins/security_solution/public/common/store/data_table/epic_local_storage.ts#L58

Reply:
should add timeout/pipe on assertion instead of delay. Because if someone changes the delay, test will start failing.

Reply:
Thanks for the help @logeekal! https://github.com/elastic/kibana/pull/156702/commits/a70dd0cf7507e0b03c05a7d19bd1ffe839e20793